### PR TITLE
Added sessionIdService and relevant tests.

### PR DESCRIPTION
**Pull request by eyurko**

### CODE REVIEW

File: client/question/services/SessionIdService.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    var sessionId = null;
+
+    /**
+     * Generates a pseudorandom sessionId.
+     *
+     * @returns {String}
+     */
+    var _createSessionId = function() {
+      var randomNumbers = new Uint32Array(SESSION_ID_LENGTH);
+      window.crypto.getRandomValues(randomNumbers);
```
Comment (seanlip) [inline client/question/services/SessionIdService.js]: May not be well-supported in IE < 11: https://developer.mozilla.org/en-US/docs/Web/API/Window/crypto

File: client/question/services/SessionIdServiceSpec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+describe('SessionIdService', function() {
+  var SessionIdService;
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    SessionIdService = $injector.get(
+      'SessionIdService');
+  }));
+
+  describe("getOrCreateSessionId", function() {
```
Comment (seanlip) [inline client/question/services/SessionIdServiceSpec.js]: nit: single quotes

File: client/question/services/SessionIdService.js (hunk 1 above)
Comment (eyurko) [reply client/question/services/SessionIdService.js]: welp

File: client/question/services/SessionIdServiceSpec.js (hunk 2 above)
Comment (eyurko) [reply client/question/services/SessionIdServiceSpec.js]: fixed

**Comment by seanlip** [comment]
LGTM pending @rabidbit's approval. Thanks!

**Comment by seanlip** [comment]
@eyurko FYI -- need to merge develop into this branch before it can be merged into master, to overcome the travis failure.

**Pull request merged**